apps/extension/src/Approvals/ApproveBond/ConfirmLedgerBond.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useState } from "react";
+import { toBase64 } from "@cosmjs/encoding";
+import BigNumber from "bignumber.js";
+
+import { LedgerError } from "@anoma/ledger-namada";
+import { Button, ButtonVariant } from "@anoma/components";
+import { defaultChainId as chainId } from "@anoma/chains";
+
+import { Ledger } from "background/ledger";
+import {
+  GetBondBytesMsg,
+  GetRevealPKBytesMsg,
+  SubmitSignedBondMsg,
+  SubmitSignedRevealPKMsg,
+} from "background/ledger/messages";
+import { Ports } from "router";
+import { closeCurrentTab } from "utils";
+import { useRequester } from "hooks/useRequester";
+import { Status } from "Approvals/Approvals";
+import {
+  ApprovalContainer,
+  ButtonContainer,
+} from "Approvals/Approvals.components";
+import { InfoHeader, InfoLoader } from "Approvals/Approvals.components";
+import { Message, RevealPKProps, Tokens } from "@anoma/types";
+import { SubmitRevealPKMsgSchema, RevealPKMsgValue } from "@anoma/types";
+import { QueryPublicKeyMsg } from "background/keyring";
+
+type Props = {
+  address: string;
+  msgId: string;
+  publicKey: string;
+};
+
+export const ConfirmLedgerBond: React.FC<Props> = ({
+  address,
+  msgId,
+  publicKey,
+}) => {
+  const requester = useRequester();
+  const [error, setError] = useState<string>();
+  const [status, setStatus] = useState<Status>();
+  const [statusInfo, setStatusInfo] = useState("");
+
+  const revealPk = async (ledger: Ledger, publicKey: string): Promise<void> => {
+    const revealPKArgs: RevealPKProps = {
+      tx: {
+        token: Tokens.NAM.address || "",
+        feeAmount: new BigNumber(0),
+        gasLimit: new BigNumber(0),
+        chainId,
+        publicKey,
+      },
+      publicKey,
+    };
+
+    const revealTxValue = new RevealPKMsgValue(revealPKArgs);
+    const txMsg = new Message<RevealPKMsgValue>();
+    const serializedTx = txMsg.encode(SubmitRevealPKMsgSchema, revealTxValue);
+
+    try {
+      const { bytes, path } = await requester.sendMessage(
+        Ports.Background,
+        new GetRevealPKBytesMsg(toBase64(serializedTx))
+      );
+
+      // Sign with Ledger
+      const signatures = await ledger.sign(bytes, path);
+
+      // Submit signatures for tx
+      setStatusInfo("Submitting reveal pk tx...");
+      await requester.sendMessage(
+        Ports.Background,
+        new SubmitSignedRevealPKMsg(msgId, toBase64(bytes), signatures)
+      );
+    } catch (e) {
+      console.warn(e);
+      throw new Error(`${e}`);
+    }
+  };
+
+  const queryPublicKey = async (
```

Comment:
Same with this function - this should apply to all Ledger tx and not need to be duplicated.